mountpoint-s3-client/src/s3_crt_client.rs

```diff
@@ -744,6 +747,42 @@ impl S3CrtClientInner {
     }
 }
 
+/// S3 operation supported by this client.
+#[derive(Debug, Clone, Copy)]
+enum S3Operation {
+    DeleteObject,
+    GetObject,
+    GetObjectAttributes,
+    HeadBucket,
+    HeadObject,
+    ListObjects,
+    PutObject,
+}
+
+impl S3Operation {
```

Comment:
why aren't there unit tests verifying that we return the right operation names and request types?

Reply:
I'm not convinced unit tests of these 2 methods would add any value. On the other hand, we do have integration tests that will use the results to initiate S3 requests.

Reply:
also thought about a unit test that verifies that for each variant of `S3Operation`, which results in `MetaRequestType::Default`, there is a not-empty `operation_name`  

but can't think of a way of writing such a test in a good way, and in the worst case (if we ever decide to use another s3 operation with default meta-request, but with no `operation_name`) we'll receive descriptive (hopefully) error from crt on request creation and catch it in a development phase